src/resources/Search/Search.ts

```diff
@@ -13,9 +13,16 @@ export default class Search extends Ressource {
     // See: https://platform.cloud.coveo.com/docs?api=SearchApi#!/Search/post_rest_search_v2
     // or : https://docs.coveo.com/en/13/cloud-v2-api-reference/search-api#operation/searchUsingGet
     query(restQueryParameters: any) {
-        return this.api.post<any>(
-            this.buildPath(Search.baseUrl, {organizationId: this.api.organizationId}),
-            restQueryParameters
-        );
+        const {viewAllContent, ...bodyParameters} = restQueryParameters;
+
+        const queryParameters = {
+            organizationId: this.api.organizationId,
+            viewAllContent: undefined,
+        };
+        if (typeof viewAllContent !== 'undefined') {
+            queryParameters.viewAllContent = viewAllContent ? 1 : 0;
+        }
```

Comment:
If we had a proper type definition for `restQueryParameters`, perhaps we wouldn't need to check if `viewAllContent` is defined or not.

We could then just do:
`queryParameters.viewAllContent = viewAllContent ? 1 : 0;`

or even inside the variable declaration:

```
        const queryParameters = {
            organizationId: this.api.organizationId,
            viewAllContent: viewAllContent ? 1 : 0,
        };
```

I guess it's a question if we care to support `viewAllContent: undefined` going forward?

Reply:
The idea is to add `viewAllContent` to the query string only when it is specified. But we could change it to add the `viewAllContent` query string argument only when it is `true`. We would get the exact same behavior in the end.

The advantage I see in checking for `undefined` is when the function is called like this:

    search.query({ viewAllContent: false });

In this case, the query string would contain `viewAllContent=0` which matches the developer intent. Omitting the query string parameter in this case might look like a bug.

Reply:
All right, I was curious but I'm fine leaving this for clarity

Reply:
Now that typing have been specifically for the `viewAllContent` query string argument, I slightly modified this part. Finally, I simplified the code to add the query string argument only when passing `{ viewAllContent: true }`.